### PR TITLE
Model `BidiAuxiliaryProperties` as an `EnumeratedProperty`

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by younies — DISMISSED** [review]

File: components/properties/src/bidi.rs [hunk 1]
```diff
@@ -2,216 +2,68 @@
...
-        let mirroring_glyph = bidi_aux_props.mirroring_glyph;
-        let paired_bracket_type = bidi_aux_props.paired_bracket_type;
-        match paired_bracket_type {
-            CheckedBidiPairedBracketType::Open => BidiPairingProperties::Open(mirroring_glyph),
-            CheckedBidiPairedBracketType::Close => BidiPairingProperties::Close(mirroring_glyph),
-            _ => BidiPairingProperties::None,
-        }
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        let [a, b, c] = unaligned.0;
+        TrieValue::try_from_u32(u32::from_le_bytes([a, b, c, 0])).unwrap_or_default()
```
Comment (Manishearth) [inline components/properties/src/bidi.rs:43]: hm. a non-safety invariant of ZeroVec is that byte equality is equality, are we okay with giving that up here?

The worst that can happen is that `==` on `ZeroVec` will do unexpected things with bad data, but I'm overall fine with that. Either way, I think we should document that this breaks the invariant but is convenient

cc @sffc

**Review by Manishearth — DISMISSED** [review]

File: components/properties/src/bidi.rs (hunk 1 above)
Comment (robertbastian) [reply components/properties/src/bidi.rs:43]: Why do you think this is being given up here?

Comment (Manishearth) [reply components/properties/src/bidi.rs:43]: The `unwrap_or_default()` means that two unequal byte sequences will produce equal values

**Review by Manishearth — APPROVED** [review]

**Comment by robertbastian** [comment]
#3573

**Pull request merged**